src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1201,16 +1204,16 @@ void Connection::receive_pong(milliseconds_type timestamp)
 
     bool legal_at_this_time = (m_waiting_for_pong && !m_send_ping);
     if (REALM_UNLIKELY(!legal_at_this_time)) {
-        logger.error("Illegal message at this time");
-        std::error_code ec = ClientError::bad_message_order;
-        close_due_to_protocol_error(ec); // Throws
+        close_due_to_protocol_error(
+            {ErrorCodes::SyncProtocolInvariantFailed, "Received PONG message when it was not valid"}); // Throws
         return;
     }
 
     if (REALM_UNLIKELY(timestamp != m_last_ping_sent_at)) {
-        logger.error("Bad timestamp in PONG message");
-        std::error_code ec = ClientError::bad_timestamp;
-        close_due_to_protocol_error(ec); // Throws
+        close_due_to_protocol_error(
+            {ErrorCodes::SyncProtocolInvariantFailed,
+             util::format("Received PONG message with an invalid timestamp (expected %1, received %2",
```

Comment:
Are we logging these messages anywhere? They would be valuable for debugging.

Reply:
Yeah, we print them in close_due_to_protocol_error now.